pkg/controller/controller_utils.go

```diff
@@ -881,7 +900,17 @@ func (s ActivePodsWithRanks) Less(i, j int) bool {
 	}
 	// 8. Empty creation time pods < newer pods < older pods
 	if !s.Pods[i].CreationTimestamp.Equal(&s.Pods[j].CreationTimestamp) {
-		return afterOrZero(&s.Pods[i].CreationTimestamp, &s.Pods[j].CreationTimestamp)
+		if s.Now.IsZero() || s.Pods[i].CreationTimestamp.IsZero() || s.Pods[j].CreationTimestamp.IsZero() {
+			return afterOrZero(&s.Pods[i].CreationTimestamp, &s.Pods[j].CreationTimestamp)
+		}
+		rankDiff := logarithmicRankDiff(s.Pods[i].CreationTimestamp, s.Pods[j].CreationTimestamp, s.Now)
+		if rankDiff < 0 {
+			return true
+		}
+		if rankDiff > 0 {
+			return false
+		}
+		return s.Pods[i].UID < s.Pods[j].UID
```

Comment:
The logic is identical for both pieces, how about making this a function rather than repeating the code?

Reply:
The idea is to keep the UID comparison in this method, for visibility. But a shorter alternative would be

```golang
diff := logarithmicRankDiff(s.Pods[i].CreationTimestamp, s.Pods[j].CreationTimestamp, s.Now)
if diff == 0 {
  return s.Pods[i].UID < s.Pods[j].UID
}
return diff < 0
```

Reply:
Yeah, I did it my way to keep the UID comparison physically last since it has the lowest priority. But if this is cleaner, we can do that too